src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -692,7 +692,7 @@ private BoundExpression BindDeclarationExpression(DeclarationExpressionSyntax no
         /// </summary>
         private BoundExpression BindDeclarationVariables(TypeSymbolWithAnnotations declType, VariableDesignationSyntax node, CSharpSyntaxNode syntax, DiagnosticBag diagnostics)
         {
-            declType = declType.IsNull ? TypeSymbolWithAnnotations.Create(CreateErrorType("var")) : declType;
+            declType = declType.IsDefault ? TypeSymbolWithAnnotations.Create(CreateErrorType("var")) : declType;
```

Comment:
>IsDefault [](start = 32, length = 9)

When `IsDefault` is true, it is technically still possible for `TypeSymbol` to be null. Maybe this should test for `HasType` instead? (same comment applies to other locations below)